### PR TITLE
refactor: 78 replace node sass with emotion

### PR DESCRIPTION
**Pull request by griffithtp**
**Type:** Refactor

The following has been addressed in the PR:

*  There is a related issue - #78 
*  Unit or Functional tests are included in the PR

**Description:**
There is 2 CSS libraries living in our code base and this PR is removing node-sass library in favor of emotion v9.
Note: there are some thoughts to move to Material UI Styles
https://github.com/verdaccio/ui/issues/78#issuecomment-508325094

**Resolves** #78

### CODE REVIEW

**Comment by juanpicado** [comment]
Awesome @griffithtp ! If there is nothing else remaining I guess we can remove `node-sass`, `sass-loader`, `stylelint-scss`, `stylelint-config-recommended-scss` from `package.json` and from the lock file.

File: src/App/styles.ts
Comment (priscilawebdev) [inline src/App/styles.ts]: I have seen that in some places you are using template literals and in other places an object ... I think we should decide which one to use and follow a standard

Comment (griffithtp) [reply src/App/styles.ts]: Glad you raise this point :)
I don't have a strong opinion on this.
Template literals feels more natural to me because it is closer to scss with simple styles. Object may be more optimal? when parsing a function.
Thoughts?

Comment (priscilawebdev) [reply src/App/styles.ts]: Hi.Thank you for asking..I would rather use objects ... since we can benefit from autocompletion and since we plan to migrate to material-ui/styles. 😉

Comment (juanpicado) [reply src/App/styles.ts]: Objects I think would be the way (not 100% sure), but for this PR template literals are more than enough, we use them all over the place.

**Comment by juanpicado** [comment]
We have small issue, the fonts are not properly applied. 
Screen Shot 2019-07-07 at 12 39 27 PM

File: src/App/styles.ts
Comment (griffithtp) [reply src/App/styles.ts]: I have updated the simple ones below.
Perhaps a slow transition as more and more Material UI components gets added?

**Comment by juanpicado** [comment]
The issue might be we were applying some styles to body check https://github.com/verdaccio/ui/pull/85/files#diff-c4de5453df6c069e26c222e36ee93651

**Review by juanpicado — CHANGES_REQUESTED** [review]
Let me know wether I can assist you in something else

File: src/App/styles.ts
Comment (griffithtp) [reply src/App/styles.ts]: fyi, have reverted again. I can't find a way to remove the compilation error it throws for some odd reasons if all exports are css objects
```
[0] src/components/Login/styles.ts
[0]  1:1  ✖  Unexpected empty source   no-empty-source
```

Comment (juanpicado) [reply src/App/styles.ts]: @griffithtp switch `no-empty-source` to warning for now in the `.eslintrc` folder, we will investigate that later on.

**Comment by griffithtp** [comment]
@priscilawebdev @juanpicado, is this redundant since moving to Material-UI?
sad to have wasted time :'(

**Comment by juanpicado** [comment]
It is fine, emotion is the main library now. Material styles is a good option but that requires a lot of research before move there. Your PR is highly valuable. 🤘🏼

**Comment by juanpicado** [comment]
I also detected a difference with the layout, this PR vs the URL below. 
Kapture 2019-07-07 at 21 18 54

https://registry.verdaccio.org/-/web/detail/webpack

**Comment by griffithtp** [comment]
thanks @juanpicado, should be fixed now

**Comment by juanpicado** [comment]
thank you @griffithtp I'll review it now

**Comment by juanpicado** [comment]
@griffithtp please read this
https://github.com/verdaccio/ui/pull/90#issuecomment-509116844
I need to hold these unfortunately.

**Comment by griffithtp** [comment]
> @griffithtp please read this
> [#90 (comment)](https://github.com/verdaccio/ui/pull/90#issuecomment-509116844)
> I need to hold these unfortunately until I verify the source of the issues.

Please let know if there is anything else to do or test here? 😃

**Comment by juanpicado** [comment]
I will @griffithtp , I just want to be sure I don't report to you fix bugs you did not created. :-) it will be unfair.

**Review by juanpicado — APPROVED** [review]
:-) 🔍 juan seal of approval 💯 . Nice work, bye bye Sass

File: src/App/App.tsx
Comment (juanpicado) [inline src/App/App.tsx]: Opps @griffithtp could we remove this?

**Review by juanpicado — CHANGES_REQUESTED** [review]
I just saw this, sorry.

File: src/App/App.tsx
Comment (priscilawebdev) [reply src/App/App.tsx]: After that I can also approve...

File: src/App/styles.ts
Comment (priscilawebdev) [inline src/App/styles.ts]: I don't use "!important;" anymore..I think when you need to use it...something is wrong..

Comment (griffithtp) [reply src/App/styles.ts]: Completely agree with you on this. 
This was a copy from the scss files I didn't want to mix up with actually refactoring styles. I think this most likely will be canned with Material-UI?

Comment (sergiohgz) [reply src/App/styles.ts]: In case we need this, we can surround in `&&` block to repeat the css class, making it important without explicitly writing the taboo word (!important)
```css
&& {
  background-color: ${colors.red};
}
```

File: src/App/App.tsx
Comment (griffithtp) [inline src/App/App.tsx]: Also removed here since no longer set in AppState.
I'm thinking in this case it might be good to use the `setLoggedUser()` as the single function to handle username, token, etc. therefore move the `storage.setItem()`?
Perhaps a different PR without deviating too much from this one 😄

File: src/App/styles.ts
Comment (sergiohgz) [reply src/App/styles.ts]: Ouch, I saw `!important` in more files, so we can keep it for a while and then remove, it isn't in the scope of the PR (great work, I'm reviewing :wink:)

File: src/App/App.tsx
Comment (juanpicado) [reply src/App/App.tsx]: For now it is ok, agree with you, a new PR but I will read that logic again later on.

**Review by sergiohgz — APPROVED** [review]
:1st_place_medal: :clap::clap::clap::clap:

File: src/App/styles.ts
Comment (juanpicado) [reply src/App/styles.ts]: Agree, I dislike important, which sounds funny. But, it was there already so, cool for now. Legacy stuff.

**Review by juanpicado — APPROVED** [review]
🤘🏼

**Comment by juanpicado** [comment]
Sergio, for some reason workflow does not get reported in PR, this is odd, sometimes does and sometimes does not. :(

**Comment by sergiohgz** [comment]
> Sergio, for some reason workflow does not get reported in PR, this is odd, sometimes does and sometimes does not. :(

@juanpicado It seems that external PR are not running the workflow in CircleCI, I will investigate how we did in other repos

**Comment by sergiohgz** [comment]
I've just enabled forked PR execution on CircleCI, it was disabled. @griffithtp, can you push an empty commit to try (if you want, if not, I'll fork and create a fake PR to test this behaviour :wink:)

circleci_status_after_enable

cc: @juanpicado

**Comment by sergiohgz** [comment]
Wow, it works, kudos @juanpicado for the advice and @griffithtp for the test :blush:

**Comment by juanpicado** [comment]
@sergiohgz I think we might have the same issue in other repos.

@griffithtp awesome, thanks so much !!

**Pull request merged**
